ygot/diff_test.go

```diff
@@ -1433,6 +1433,53 @@ func TestDiff(t *testing.T) {
 		inOrig:        &renderExample{},
 		inMod:         &pathElemExample{},
 		wantErrSubStr: "cannot diff structs of different types",
+	}, {
+		desc:   "prepend module",
+		inOrig: &ietfRenderExample{F2: String("f2")},
+		inMod: &ietfRenderExample{
+			F1: String("f1"),
+			F3: &ietfRenderExampleChild{F5: String("f5")},
+		},
+		inOpts: []DiffOpt{&PrependModuleNames{}},
+		want: &gnmipb.Notification{
+			Delete: []*gnmipb.Path{{
+				Elem: []*gnmipb.PathElem{{
+					Name: "f2mod/f2mod:config",
+				}, {
+					Name: "f2mod/f2mod:f2",
+				}},
```

Comment:
Thanks for the quick review - I'll aim to address then in the next day or two :+1:

Reply:
Sorry @wenovus for the delay here :disappointed: 

> The reason is the struct tag "f2mod/f2mod|f2mod" corresponds one-to-one with "config/f2|f2mod".

I think you might be looking at the wrong test struct. In the test I'm using [`ietfRenderExample{}`](https://github.com/openconfig/ygot/blob/f50653f3a4f2755cb319df362b9b6253453533eb/ygot/render_test.go#L1901-L1909) where the `module` struct tag on field F2 is just "f2mod/f2mod".

Related question, and I'm a bit new to YANG and I don't know ygot deeply (yet), but I believe according to RFC6020 "f2mod/f2mod" isn't a valid module name. I see that pattern in use in the tests structs in a few places - can you explain a little more? Sorry for the rookie question.

With an answer to :point_up: that I think I can work out the rest

Reply:
When you see this in the code:
```go
F2 `path:"config/f2" module:"f2mod/f2mod"`
```
It means that the field `F2` is a [compressed field](https://github.com/openconfig/ygot/blob/master/docs/design.md#openconfig-path-compression) and its relative path from the parent struct is "config/f2". The module namespace of each of these relative path elements are "f2mod", and "f2mod" respectively, since it is possible in the general case that "config" and "f2" belong to different module namespaces.

The
```go
`path:"config/f2|f2" module:"f2mod/f2mod|f2mod"`
```
I mentioned above is another case which you will need to handle, where this struct field represents both the "config/f2" and "f2" relative paths in YANG so you need to split on `|` as I mentioned above. The reason for its existence is due to the idiosyncrasy of OpenConfig, where this is used to represent compressed leaves which are also list keys.

This is partly why I mentioned this is a pretty complex part of the code base. Let me know how I can clarify this further.